### PR TITLE
sev: update skopeo encrypt command to use 'docker' target

### LINKED ISSUE #97: quickstart: encryption with skopeo's `docker-daemon` protocol doesn't work
The quickstart contain a section on how to [create an encrypted image](https://github.com/confidential-containers/documentation/blob/main/quickstart.md#create-an-encrypted-image) with the attestation-agent. It mentions the following command to encrypt the image:

```
sudo OCICRYPT_KEYPROVIDER_CONFIG=$(pwd)/attestation-agent/sample_keyprovider/src/enc_mods/offline_fs_kbs/ocicrypt.conf \
skopeo copy --insecure-policy docker-daemon:[REGISTRY_URL]:unencrypted \
docker-daemon:[REGISTRY_URL]:encrypted \
--encryption-key provider:attestation-agent:$(pwd)/keys.json:key_id1
```

However, I realized that skopeo will silently leave the image unencrypted if the target image URL uses the `docker-daemon` protocol. In my experiments I could only generate the image properly with `docker` or `oci`. With the former `skopeo` will push the image directly to an image registry, whereas the later will create a local directory and you must use skopeo again to push it to the registry.

Cc @ryansavino @fitzthum

### PR DESCRIPTION
**Pull request by ryansavino**
Fixes: #97

Signed-Off-By: Ryan Savino <ryan.savino@amd.com>

### CODE REVIEW

File: quickstart.md [hunk 1]
```diff
@@ -759,14 +759,19 @@ Run skopeo to encrypt the image created in the previous section:
 
 ```
 sudo OCICRYPT_KEYPROVIDER_CONFIG=$(pwd)/attestation-agent/sample_keyprovider/src/enc_mods/offline_fs_kbs/ocicrypt.conf \
-skopeo copy --insecure-policy docker-daemon:[REGISTRY_URL]:unencrypted \
-docker-daemon:[REGISTRY_URL]:encrypted \
+skopeo copy --insecure-policy \
+docker-daemon:[REGISTRY_URL]:unencrypted \
```
Comment (fitzthum) [inline quickstart.md:763]: Can't we use `docker:` for the source as well?

**Review by fitzthum — APPROVED** [review]

File: quickstart.md (hunk 1 above)
Comment (wainersm) [reply quickstart.md:763]: Yes, we can:

```
$ ./skopeo copy --insecure-policy --encryption-key provider:attestation-agent:${KEYS_FILE}:key_id1 docker://nginx:latest docker://wainersm/nginx-encrypted:test
Getting image source signatures
Copying blob d2c0556a17c5 done  
Copying blob 8740c948ffd4 done  
Copying blob b2fe3577faa4 done  
Copying blob c8b9881f2c6a done  
Copying blob 8316c5e80e6d done  
Copying blob 693c3ffa8f43 done  
Copying config 75eccb489a done  
Writing manifest to image destination
Storing signatures
$ ./skopeo inspect docker://wainersm/nginx-encrypted:test | grep MIMEType
            "MIMEType": "application/vnd.oci.image.layer.v1.tar+gzip+encrypted",
            "MIMEType": "application/vnd.oci.image.layer.v1.tar+gzip+encrypted",
            "MIMEType": "application/vnd.oci.image.layer.v1.tar+gzip+encrypted",
            "MIMEType": "application/vnd.oci.image.layer.v1.tar+gzip+encrypted",
            "MIMEType": "application/vnd.oci.image.layer.v1.tar+gzip+encrypted",
            "MIMEType": "application/vnd.oci.image.layer.v1.tar+gzip+encrypted",

```

File: quickstart.md [hunk 2]
```diff
@@ -759,14 +759,19 @@ Run skopeo to encrypt the image created in the previous section:
...
 ```
 
 Again, be sure to replace `[REGISTRY_URL]` with the desired registry URL.
 `--insecure-policy` flag is used to connect to the attestation agent and will not impact the security of the project.
 
+Make sure to use the `docker` target in the destination URL in the `skopeo copy` command as demonstrated above. Publishing the 
+encrypted image to the local `docker-daemon` is known to have issues and not provide an adequate error response. It is better to 
+publish directly to a remote registry known to support encrypted images like GitHub Container Registry (GHCR).
+
 At this point it is a good idea to inspect the image was really encrypted as skopeo can silently leave it unencrypted. Use
```
Comment (wainersm) [inline quickstart.md:775]: @ryansavino you will need to rephrase that.

Comment (wainersm) [reply quickstart.md:775]: ah, and also below when you read "Push the encrypted image to the registry:"...it doesn't need to push the image to docker as skopeo does it already.

File: quickstart.md (hunk 1 above)
Comment (ryansavino) [reply quickstart.md]: Updated.

File: quickstart.md (hunk 2 above)
Comment (ryansavino) [reply quickstart.md:775]: Check it out now, and let me know if it reads okay.

**Review by wainersm — APPROVED** [review]
LGTM. Thanks @ryansavino !

**Comment by wainersm** [comment]
@confidential-containers/documentation-maintainers could you please review this PR? It is blocked waiting at least one maintainer to approve.

**Review by c3d — APPROVED** [review]
LGTM

**Pull request merged**